router/router_test.go

```diff
@@ -103,32 +108,29 @@ func TestHandshakeBadRealm(t *testing.T) {
 	}
 	defer r.Close()
 
-	client, server := transport.LinkedPeers(0, log)
-
-	client.Send(&wamp.Hello{Realm: "does.not.exist"})
+	client, server := transport.LinkedPeers(log)
+	go client.Send(&wamp.Hello{Realm: "does.not.exist"})
 	err = r.Attach(server)
-	if err == nil {
-		t.Error(err)
-	}
-
-	if len(client.Recv()) != 1 {
-		t.Fatal("Expected one message in the handshake, received ",
-			len(client.Recv()))
+	if err != nil {
+		t.Fatal(err)
 	}
 
-	msg := <-client.Recv()
-	if msg.MessageType() != wamp.ABORT {
-		t.Error("Expected ABORT after handshake")
+	select {
+	case <-time.After(time.Second):
+		t.Fatal("timed out waiting for response to HELLO")
+	case msg := <-client.Recv():
+		if _, ok := msg.(*wamp.Abort); !ok {
+			t.Error("Expected ABORT after bad handshake")
+		}
 	}
 }
 
 func TestRouterSubscribe(t *testing.T) {
 	const testTopic = wamp.URI("some.uri")
-
-	sub, subServer := transport.LinkedPeers(0, log)
-	r, err := newTestRouter()
+	r, _ := newTestRouter()
```

Comment:
even though this generally should not fail in the test, I think it is a good idea not to ignore the err (if there is one)